vm/src/frame.rs

```diff
@@ -157,6 +158,16 @@ impl NameProtocol for Scope {
         None
     }
 
+    fn store_cell(&self, vm: &VirtualMachine, name: &str, value: PyObjectRef) {
+        self.locals
+            .iter()
+            .skip(1)
+            .next()
+            .unwrap()
```

Comment:
This panic here is an interesting case. It looks like cpython raises ```SyntaxError``` is it can't find the relevant variable at compile time. You can't create a variable in an outer scope with ```nonlocal```, only change it. (Unlike ```global```.)

We can't do better than this yet, but worth using ```.expect("no outer scope for non-local")```?

Reply:
Allright, good suggestion! A next topic for me would be to detect more errors at compile time, like cpython does, and raise proper syntax errors.